dot/state/roottottrie.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/trie"
+)
+
+type rootToTrieMap struct {
```

Comment:
```suggestion
type trieMap struct {
```

Reply:
Changed it to `tries` since it now has the in memory map and a reference to the persistent database.